### PR TITLE
Add iOS tap in empty space to show toolbar, Android long-press in empty space to show toolbar. (Resolves #1472)

### LINKED ISSUE #1472: [SuperTextField][iOS/Android] Not possible to show the text selection toolbar when the field is empty
Steps to reproduce:
* Open the SuperTextField demo
* Remove all text in the field
* Tap twice on the empty field

**Expected:** the text selection toolbar is shown
**Actual:** nothing happens besides the field gaining focus

Having the ability to show the toolbar when the field is empty, is important for being able to paste text (e.g. an email during login flow) on a mobile device.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Add iOS tap in empty space to show toolbar, Android long-press in empty space to show toolbar. (Resolves #1472)

### CODE REVIEW

**Comment by matthew-carroll** [comment]
CC @miguelcmedeiros

File: super_editor/lib/src/super_textfield/ios/_user_interaction.dart [hunk 1]
```diff
@@ -156,40 +155,42 @@ class IOSTextFieldTouchInteractorState extends State<IOSTextFieldTouchInteractor
...
-      // Place the caret based on the tap offset. In this case, the caret will
-      // be placed at the end of text because the user tapped in empty space.
-      _selectAtOffset(details.localPosition);
-      return;
-    }
+    final exactTapTextPosition = _getTextPositionAtOffset(details.localPosition);
+    final didTapOnExistingSelection = exactTapTextPosition != null &&
+        _selectionBeforeTap != null &&
+        (_selectionBeforeTap!.isCollapsed
+            ? exactTapTextPosition == _selectionBeforeTap!.extent
```
Comment (angelosilvestre) [inline super_editor/lib/src/super_textfield/ios/_user_interaction.dart:162]: Should we compare just the offset to consider selections with different affinities equal?

File: super_editor/lib/src/super_textfield/ios/_user_interaction.dart [hunk 2]
```diff
@@ -156,40 +155,42 @@ class IOSTextFieldTouchInteractorState extends State<IOSTextFieldTouchInteractor
...
-        ? tapTextPosition == previousSelection.extent
-        : tapTextPosition.offset >= previousSelection.start && tapTextPosition.offset <= previousSelection.end;
+    // Select the text that's nearest to where the user tapped.
+    _selectAtOffset(details.localPosition);
 
-    if (didTapOnExistingSelection) {
-      // Toggle the toolbar display when the user taps on the collapsed caret,
-      // or on top of an existing selection.
+    final didCaretStayInSamePlace = _selectionBeforeTap != null &&
+        _selectionBeforeTap == widget.textController.selection &&
```
Comment (angelosilvestre) [inline super_editor/lib/src/super_textfield/ios/_user_interaction.dart:170]: Same question here about selection affinity.

**Review by angelosilvestre — APPROVED** [review]
LGTM with a couple comments.

**Pull request merged**